snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -355,15 +360,9 @@ def ignore(directory, files):
         shutil.copytree(self.plugin.sourcedir, self.plugin.build_basedir,
                         symlinks=True, ignore=ignore)
 
-        script_runner = ScriptRunner(builddir=self.plugin.build_basedir)
-
-        script_runner.run(scriptlet=self._part_properties.get('prepare'))
-        build_scriptlet = self._part_properties.get('build')
-        if build_scriptlet:
-            script_runner.run(scriptlet=build_scriptlet)
-        else:
-            self.plugin.build()
-        script_runner.run(scriptlet=self._part_properties.get('install'))
+        self._runner.pre_build()
```

Comment:
I am still on the edge about this, but we should probably make `pre/post-build` actual lifecycle stages that do things (what `prepare-` does today`) and have real things take place there (which could be called or setup by the script), in way:

```
pre-build: |
    snapcraft_install_stage_packages
    patch -p1 file < ...
```

and so on and so forth. Let's talk about this on Monday!

Reply:
> we should probably make pre/post-build actual lifecycle stages that do things [...]

That's exactly what I'm shooting for with [this logic](https://github.com/snapcore/snapcraft/pull/1998/files#diff-313e8dc0aa8d02f56e6ec918c25ddd93R29) to have default actions for pre/post-build, it's just keeping that logic in the runner instead of the pluginhandler, so the pluginhandler can just set it up in `__init__` and not worry about it anymore (it's pretty bloated). Does that align with the goals, here?

Reply:
From your example, perhaps you're simply wanting to rewrite our current `prepare_pull` and `prepare_build` as default `pre-pull` and `pre-build` scriptlets that calls the `snapcraft_{fetch,unpack}_stage_packages` function we define to fetch/install stage-packages. That's an interesting idea.

Reply:
let's sync up after the weekly call today